### PR TITLE
Dịch "softmax-regression-scratch_vn" - Phần 5

### LINKED ISSUE #771: Dịch "softmax-regression-scratch_vn" - Phần 5
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/softmax-regression-scratch_vn.md#L329-L453

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by di-mi-ta**
V1

## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #771

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 1]
```diff
@@ -580,7 +588,8 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 5 -->
-*
+* Đinh Minh Tân
+* Pham Hong Vinh
```
Comment (duythanhvn) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]: Sao chưa gì đã có tên rồi bạn @di-mi-ta ?

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 2]
```diff
@@ -430,7 +435,7 @@ class Animator(object):
 The training function then runs multiple epochs and visualize the training progress.
 -->
 
-*dịch đoạn phía trên*
+Hàm huấn luyện sau đó sẽ chạy qua nhiều epochs và trực quan quá trình huấn luyện.
```
Comment (MarkusKhoa) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Hàm huấn luyện sau đó sẽ chạy qua nhiều epochs và trực quan hoá quá trình huấn luyện. 
```

Comment (MarkusKhoa) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Mình nghĩ nên dùng chữ "trực quan hoá" để thể hiện rõ đây là một động từ

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 3]
```diff
@@ -358,7 +358,11 @@ Note that `updater` is general function to update the model parameters, which ac
 It can be either a wrapper of `d2l.sgd` or a Gluon trainer.
 -->
 
-*dịch đoạn phía trên*
+Việc lặp để huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua sự hiện thực cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Vòng lặp huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua cách lập trình cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 4]
```diff
@@ -358,7 +358,11 @@ Note that `updater` is general function to update the model parameters, which ac
 It can be either a wrapper of `d2l.sgd` or a Gluon trainer.
 -->
 
-*dịch đoạn phía trên*
+Việc lặp để huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua sự hiện thực cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
+Ở đây, chúng ta tái cấu trúc lại sự hiện thực để giúp nó có thể được tái sử dụng.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Ở đây, chúng ta tái cấu trúc lại đoạn mã để giúp nó có thể được tái sử dụng. 
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 5]
```diff
@@ -358,7 +358,11 @@ Note that `updater` is general function to update the model parameters, which ac
 It can be either a wrapper of `d2l.sgd` or a Gluon trainer.
 -->
 
-*dịch đoạn phía trên*
+Việc lặp để huấn luyện cho hồi quy softmax trông khá quen thuộc nếu bạn đã đọc qua sự hiện thực cho hồi quy tuyến tính tại :numref:`sec_linear_scratch`. 
+Ở đây, chúng ta tái cấu trúc lại sự hiện thực để giúp nó có thể được tái sử dụng. 
+Đầu tiên, chúng ta định nghĩa một hàm để huấn luyện cho 1 epoch dữ liệu. 
+Lưu ý rằng `updater` là một hàm tổng quát để cập nhật các tham số của mô hình và sẽ nhận giá trị kích thước batch làm thông số. 
+Nó có thể là một wrapper của `d2l.sgd` hoặc là một đối tượng huấn luyện Gluon.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Nó có thể là một hàm gói quanh `d2l.sgd` hoặc là một đối tượng huấn luyện Gluon. 
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 6]
```diff
@@ -383,7 +387,8 @@ Before showing the implementation of the training function, we define a utility
 Again, it aims to simplify the codes in later chapters.
 -->
 
-*dịch đoạn phía trên*
+Trước khi trình bày sự hiện thực của hàm huấn luyện, chúng ta định nghĩa 1 lớp phụ trợ để trực quan dữ liệu.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Trước khi xem đoạn mã thực hiện hàm huấn luyện, chúng ta định nghĩa 1 lớp phụ trợ để biểu diễn trực quan dữ liệu.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 7]
```diff
@@ -451,7 +456,10 @@ By changing their values, we may be able to increase the classification accuracy
 In practice we will want to split our data three ways into training, validation, and test data, using the validation data to choose the best values of our hyperparameters.
 -->
 
-*dịch đoạn phía trên*
+Nhắc lại, chúng ta sử dụng giải thuật hạ gradient ngẫu nhiên theo minibatch để tối ưu hàm mất mát của mô hình.
+Lưu ý rằng số lượng epochs (`num_epochs`), và hệ số học (`lr`) là 2 siêu tham số được hiệu chỉnh. 
+Bằng cách thay đổi các giá trị này, chúng ta có thể tăng độ chính xác khi phân loại của mô hình. 
+Trong thực tế, chúng ta thường sẽ chia tập dữ liệu thành 3 phần, đó là: dữ liệu huấn luyện, dữ liệu kiểm thử và dữ liệu kiểm tra, sử dụng dữ liệu kiểm thử để chọn ra những giá trị tốt nhất cho các siêu tham số.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]: Hồi mình dịch `dev set` là `tập phát triển`, giờ mình có dịch `validation set` vậy luôn không hay dịch sát nghĩa chữ `validation` nhỉ? @tiepvupsu @minhduc0711

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 1 above)
Comment (di-mi-ta) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: À sorry, mình thêm nhầm ấy! Lộn qua pull request trước! :)) Để mình sửa lại!

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 5 above)
Comment (di-mi-ta) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Chỗ này mình không biết có nên dịch wrapper ra tiếng Viêt hay không hay để tiếng Anh cho quen thuộc!

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 7 above)
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: @lkhphuc giờ mình mới biết `dev set` là từ khác cho `validation set` :smile:
Vậy mình nghĩ nên dịch cả 2 là `tập kiểm định` cho đỡ rối.

Comment (tiepvupsu) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Không hẳn đâu, mình cứ dịch `validation set` riêng đi. Hầu hết các sách dùng `validation set`, chỉ thấy cuốn MLY dùng `dev set` thôi.

Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Đúng hơn là chỉ thấy ông Andrew Ng (+học trò ổng) dùng dev set thôi.

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 5 above)
Comment (lkhphuc) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: @minhduc0711 thấy sao?

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: @lkhphuc mấy term trong lập trình thì mình cũng mù. Nếu từ tiếng việt không quá obvious thì mình nên tạm thời để nguyên.

Comment (duythanhvn) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Từ này mình không vote dịch, trong kỹ thuật lập trình từ này cũng không dịch ra vì nó giống tên mô tả của một khái niệm (trong Java từ này mô tả cơ chế chuyển đổi kiểu dữ liệu nguyên thủy thành kiểu đối tượng và ngược lại). Dịch ra nó lại .... củ chuối như trên...

Comment (tanduong) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: `wrapper` trong trường hợp này nghĩa là `sử dụng`.

```suggestion
Nó có thể là một có sử dụng `d2l.sgd` hoặc là một đối tượng huấn luyện Gluon. 
```

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: @tanduong nghe cũng hợp lý, nhưng mình nghĩ không nên thay đổi mạnh tay từ ngữ của tác giả
